api/dataservices/edgestack/edgestack.go

```diff
@@ -67,14 +67,16 @@ func (service *Service) EdgeStack(ID portainer.EdgeStackID) (*portainer.EdgeStac
 	return &stack, nil
 }
 
-// CreateEdgeStack assign an ID to a new Edge stack and saves it.
+// CreateEdgeStack saves an Edge stack object to db. If no ID was assigned, it will assign a new one.
 func (service *Service) Create(edgeStack *portainer.EdgeStack) error {
-	return service.connection.CreateObject(
+	if edgeStack.ID == 0 {
+		edgeStack.ID = portainer.EdgeStackID(service.GetNextIdentifier())
+	}
```

Comment:
The only observable difference in this code is that if ID already exists; then we don't assign one.

If this is the case, then the caller of this function should look at the ID of the edgeStack and either call `Update` or `Create` (based on presence of ID) - this ID check shouldn't be done in the `Create` function (and hence we should probably leave this function as is - but change the caller instead)?

Im not sure if i'm missing something here.

Reply:
I thought of changing the signature to `Create(id, stack)` just to make it explicit. wdyt?

we need the id before creation to save the stacks files in its folder

Reply:
Ah ok i see; this makes more sense.